src/vote_graph.rs

```diff
@@ -119,6 +119,49 @@ impl<H, N, V> VoteGraph<H, N, V> where
 		(self.base.clone(), self.base_number)
 	}
 
+	/// Adjust the base of the graph. The new base must be an ancestor of the
+	/// old base.
+	///
+	/// Provide an ancestry proof from the old base to the new. The proof
+	/// should be in reverse order from the old base's parent.
+	pub fn adjust_base(&mut self, ancestry_proof: &[H]) {
+		let new_hash = match ancestry_proof.last() {
+			None => return, // empty -- nothing to do.
+			Some(h) => h,
+		};
+
+		// not a valid ancestry proof. TODO: error?
+		if ancestry_proof.len() > self.base_number.as_() { return }
+
+		// hack because we can't convert usize -> N, only vice-versa.
+		// hopefully LLVM can optimize.
```

Comment:
Maybe we could add `From<u64>` to `BlockNumberOps`.

Reply:
I wanted to avoid that because it would probably cause issues with `u32` block numbers (very likely in ephemeral chains).  A `TryFrom` implementation would  be fine. i'll add a TODO.